sklearn/compose/tests/test_column_transformer.py

```diff
@@ -1180,3 +1184,83 @@ def test_column_transformer_mask_indexing(array_type):
     )
     X_trans = column_transformer.fit_transform(X)
     assert X_trans.shape == (3, 2)
+
+
+@pytest.mark.parametrize('cols,pattern,include,exclude', [
+    (['col_int', 'col_float'], None, np.number, None),
+    (['col_int', 'col_float'], None, None, object),
+    (['col_int', 'col_float'], None, [np.int, np.float], None),
+    (['col_str'], None, [np.object], None),
+    (['col_str'], None, np.object, None),
+    (['col_float'], None, float, None),
+    (['col_float'], 'at$', [np.number], None),
+    (['col_int'], None, [np.int], None),
+    (['col_int'], '^col_int', [np.number], None),
+    (['col_float', 'col_str'], 'float|str', None, None),
+    (['col_str'], '^col_s', None, [np.int]),
+    (['col_int', 'col_float', 'col_str'], None, [np.number, np.object], None),
```

Comment:
Add a test where none of the conditions are met? That should just return [] right?